src/image_embedding/impl.rs

```diff
@@ -132,6 +132,97 @@ impl ImageEmbedding {
             .expect("Model not found.")
     }
 
+    /// Method to generate image embeddings for a Vec of image bytes
+    pub fn embed_bytes(
+        &self,
+        images: &[&[u8]],
+        batch_size: Option<usize>,
+    ) -> anyhow::Result<Vec<Embedding>> {
+        let batch_size = batch_size.unwrap_or(DEFAULT_BATCH_SIZE);
```

Comment:
There's quite a lot of code duplication from the embed() method here. Maybe refactor?

Reply:
Should be good now!

I noticed some tests are failing but that was even before the refactor. I'm assuming that's unrelated to this PR